tile.cpp

```diff
@@ -633,13 +633,18 @@ static double simplify_feature(serial_feature *p, drawvec const &shared_nodes, n
 			}
 
 			if (!already_marked) {
-				if (p->coalesced && t == VT_POLYGON) {
+				if (p->coalesced) {
 					// clean coalesced polygons before simplification to avoid
 					// introducing shards between shapes that otherwise would have
-					// unioned exactly
-					//
-					// don't try to scale up because these are still world coordinates
-					geom = clean_or_clip_poly(geom, 0, 0, false, false);
+					// unioned exactly.
+					if (t == VT_POLYGON) {
+						// don't scale up since this is still world coordinates
+						geom = coalesce_polygon(geom, false);
+					} else if (t == VT_LINE) {
+						geom = coalesce_linestring(geom);
+					}
+
+					p->coalesced = false;
```

Comment:
the flag "coalesced" is somewhat confusing to me here, because it's really something like "needs cleaning" (which is false after running this code).